File: networking/src/main/java/net/remmintan/mods/minefortress/networking/c2s/C2SClearActiveBlueprint.java
package net.remmintan.mods.minefortress.networking.c2s;

import net.minecraft.network.PacketByteBuf;
import net.minecraft.server.MinecraftServer;
import net.minecraft.server.network.ServerPlayerEntity;
import net.remmintan.mods.minefortress.core.interfaces.server.IFortressServer;
import net.remmintan.mods.minefortress.networking.interfaces.FortressC2SPacket;

public class C2SClearActiveBlueprint implements FortressC2SPacket {

    public static final String CHANNEL = "clear_active_blueprint";

    public C2SClearActiveBlueprint() {
    }
    public C2SClearActiveBlueprint(PacketByteBuf buf) {
    }

    @Override
    public void handle(MinecraftServer server, ServerPlayerEntity player) {
        var fortressServer = (IFortressServer) server;
        fortressServer.get_BlueprintsWorld().clearBlueprint(player);
    }

    @Override
    public void write(PacketByteBuf buf) {

    }
}

File: src/networking/java/net/remmintan/mods/minefortress/networking/c2s/C2SClearActiveBlueprint.kt
package net.remmintan.mods.minefortress.networking.c2s

import net.minecraft.network.PacketByteBuf
import net.minecraft.server.MinecraftServer
import net.minecraft.server.network.ServerPlayerEntity
import net.remmintan.mods.minefortress.core.interfaces.networking.FortressC2SPacket
import net.remmintan.mods.minefortress.core.interfaces.server.IFortressServer

class C2SClearActiveBlueprint : FortressC2SPacket {
    constructor()
    constructor(ignoredBuf: PacketByteBuf?)

    override fun handle(server: MinecraftServer, player: ServerPlayerEntity) {
        if (server is IFortressServer) {
            server._BlueprintWorld.clearBlueprint(player)
        }
    }

    override fun write(buf: PacketByteBuf) {
    }

    companion object {
        const val CHANNEL: String = "clear_active_blueprint"
    }
}
